admin/howto/environment.md

```diff
@@ -5,7 +5,92 @@ configured environment with base libraries, user interfaces and
+   Here are a few good resources for defining these files:
+
+   - The Binder Project has [documentation about what files are supported](https://mybinder.readthedocs.io/en/latest/using/config_files.html).
+   - The Turing Way has an excellent a collection of tutorials for getting started with Binder, and covers [Python](https://github.com/alan-turing-institute/the-turing-way/blob/main/workshops/boost-research-reproducibility-binder/workshop-presentations/zero-to-binder-python.md), [Julia](https://github.com/alan-turing-institute/the-turing-way/blob/main/workshops/boost-research-reproducibility-binder/workshop-presentations/zero-to-binder-julia.md), and [the R language](https://github.com/alan-turing-institute/the-turing-way/blob/main/workshops/boost-research-reproducibility-binder/workshop-presentations/zero-to-binder-r.md).
+3. **Create a Docker image registry account**. This will be the place where you store user images once they are built, so that the hub can access them.
+   :::{tip}
+   We recommend using [the quay.io image registry](https://quay.io/) for your Docker images.
+   This is a public registry service run by [Red Hat](https://www.redhat.com/en), and is reliable to use.
+   :::
+4. **Set up a GitHub Action** to automatically build a Docker image using [the repo2docker action](https://github.com/jupyterhub/repo2docker-action), and push it to a registry.
+5. **Configure your hub** to pull the user image from the registry above, either via [the configurator](configurator.md) (to do it yourself) or by [opening a support ticket](../../support.md) (to request that a 2i2c engineer do this for you).
+
+:::{tip}
+By following these steps, you have also created a [Binder-ready repository](https://mybinder.org), and we recommend trying to build your reposity on mybinder.org in order to test things out.
+:::
+
+#### An in-depth guide
+
+To help you get started, we've created an [environment generation guide](https://github.com/2i2c-org/hub-user-image-template/blob/main/README.md) and a small template repository that will guide you through the process.
+Go to the repository by clicking the button below, and follow the instructions in the README for next steps.
+
+```{button-link} https://github.com/2i2c-org/hub-user-image-template/blob/main/README.md
+:color: primary
+Go to user environment template
+```
+
+### Temporarily install packages for a session
+
+You can temporarily install packages in your environment that will
+just last the duration of your user session. They will get wiped out
+when your user server is stopped, to ensure that you always start from
+a 'clean slate' environment.
+
+The recommended way is to put `%pip install <list-of-packages>` or
+`%conda install <list-of-packages>` in the first cell of any notebook
+you distribute, so when run it'll install necessary packages. For R,
+you can use `install.packages("package-name")` as you normally would.
+
+```{warning}
```

Comment:
What's the difference between the `%` and `!` prefixes in the 2 `pip` examples? Do they cause the packages to be installed in different places, or is that solely the `--user` flag?

(This is probably more for my own understanding than a requirement for the docs?)